database/buildSchema/28_review_assignment.sql

```diff
@@ -42,13 +45,36 @@ CREATE TABLE public.review_assignment (
     is_self_assignable boolean DEFAULT FALSE
 );
 
--- TRIGGER (Listener) on review_assignment table
+-- FUNCTION
+CREATE OR REPLACE FUNCTION public.unassign_review_without_sections ()
+    RETURNS TRIGGER
+        AS $review_assignment_event$
+    BEGIN
+        UPDATE
+            public.review_assignment
+        SET
+            status = 'AVAILABLE'
+        WHERE
+            id = NEW.id;
+        RETURN NULL;
+    END;
+$review_assignment_event$
+LANGUAGE plpgsql;
```

Comment:
Our assignment mutations *should* be setting this correctly, right? So in theory this shouldn't need to execute.

Or are you just changing the sections in the front-end when un-assigning and leaving the "status" field to this trigger function?

Reply:
Sorry - this work is partially done. II would like to give this a better solution in a PR to fix [this](https://github.com/openmsupply/application-manager-web-app/issues/1109) Front-end issue. But yes, the idea is that the Front-end would send the reassigned sections to the reassign mutation and keep the status as ASSIGNED if there is still one. Or set the status to AVAILABLE when all sections have been un-assigned. But since this is a front-end solution I thought it would be good to add a safe guard here.
There is a problem still with the trigger, because the front-end would be setting trigger: ON_REVIEW_UNASSIGN when if encouters a fully unassignment reviewAssignment, but this safe-guard won't do it, so it wouldn't trigger an unassignment correctly.